src/AutoRest.CSharp/Common/Output/Models/ValueExpressions.cs

```diff
@@ -0,0 +1,291 @@
+                    {
+                        // When `JsonElement` provides access to its UTF8 buffer, change this code to create `BinaryData` from it.
+                        // See also PageableHelpers.ParseResponseForBinaryData
+                        var e = new CodeWriterDeclaration("e");
+                        return Func(e, Call.Static(BinaryDataType, nameof(BinaryData.FromString), Call.Instance(e, nameof(JsonElement.GetRawText))));
+                    }
+
+                    if (!pageItemType.IsFrameworkType && pageItemType.Implementation is SerializableObjectType { JsonSerialization: { }, IncludeDeserializer: true } type)
+                    {
+                        return new MemberReference(new TypeReference(type.Type), $"Deserialize{type.Declaration.Name}");
+                    }
+
+                    var deserializeImplementation = new FormattableStringToExpression(JsonCodeWriterExtensions.GetDeserializeValueFormattable($"e", pageItemType));
+                    return Func(new CodeWriterDeclaration("e"), deserializeImplementation);
+                }
+            }
+
+            public static class ProtocolOperationHelpers
+            {
+                public static ValueExpression ProcessMessage(CodeWriterDeclaration pipeline, CodeWriterDeclaration message, CodeWriterDeclaration clientDiagnostics, string scopeName, OperationFinalStateVia finalStateVia, bool async)
+                {
+                    var methodName = async ? nameof(Azure.Core.ProtocolOperationHelpers.ProcessMessageAsync) : nameof(Azure.Core.ProtocolOperationHelpers.ProcessMessage);
+                    return ProcessMessage(pipeline, message, clientDiagnostics, scopeName, finalStateVia, async, methodName);
+                }
+
+                public static ValueExpression ProcessMessageWithoutResponseValue(CodeWriterDeclaration pipeline, CodeWriterDeclaration message, CodeWriterDeclaration clientDiagnostics, string scopeName, OperationFinalStateVia finalStateVia, bool async)
+                {
+                    var methodName = async ? nameof(Azure.Core.ProtocolOperationHelpers.ProcessMessageWithoutResponseValueAsync) : nameof(Azure.Core.ProtocolOperationHelpers.ProcessMessageWithoutResponseValue);
+                    return ProcessMessage(pipeline, message, clientDiagnostics, scopeName, finalStateVia, async, methodName);
+                }
+
+                private static ValueExpression ProcessMessage(CodeWriterDeclaration pipeline, CodeWriterDeclaration message, CodeWriterDeclaration clientDiagnostics, string scopeName, OperationFinalStateVia finalStateVia, bool async, string methodName)
+                {
+                    var arguments = new List<ValueExpression> {
+                        new VariableReference(pipeline),
+                        new VariableReference(message),
+                        new VariableReference(clientDiagnostics),
+                        new FormattableStringToExpression($"{scopeName:L}"),
```

Comment:
Should we use the `Literal` method here?

Reply:
Yes